src/filterTargets.ts

```diff
@@ -1,23 +1,27 @@
-"use strict";
+import { intersection, difference } from "lodash";
+import { FirebaseError } from "./error";
+import { Options } from "./options";
 
-var _ = require("lodash");
-var { FirebaseError } = require("./error");
-
-module.exports = function (options, validTargets) {
-  var targets = validTargets.filter(function (t) {
+/**
+ * Filters targets from options with valid targets as specified.
+ * @param options CLI options.
+ * @param validTargets Targets that are valid.
+ * @return List of targets as specified and filtered by options and validTargets.
+ */
+export function filterTargets(options: Options, validTargets: string[]): string[] {
+  let targets = validTargets.filter((t) => {
     return options.config.has(t);
   });
   if (options.only) {
-    targets = _.intersection(
+    targets = intersection(
       targets,
-      options.only.split(",").map(function (opt) {
+      options.only.split(",").map((opt: string) => {
         return opt.split(":")[0];
       })
     );
   } else if (options.except) {
-    targets = _.difference(targets, options.except.split(","));
+    targets = difference(targets, (options.except || "").split(","));
```

Comment:
Since this is inside a `if (options.except)`, this `|| ""` seems redundant.

Reply:
I think something was yelling at me before about it maybe being undefined. Looks like I can remove it now. Thanks.